### PR TITLE
Twig component: Makes breadcrumbs management independent

### LINKED ISSUE #33603: Independent breadcrumbs twig component
A bit more details in #33182

### PR DESCRIPTION
**Pull request by M0rgan01**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Makes breadcrumbs management independent
| Type?             | improvement
| Category?         |BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI 🟢 and auto tests 🟢
| Fixed issue or discussion?     | Fixes #33603
| Related PRs       |  -
| Sponsor company   | -

### CODE REVIEW

**Comment by M0rgan01** [comment]
Tests: https://github.com/M0rgan01/ga.tests.ui.pr/actions/runs/5939492195

**Review by nicosomb — DISMISSED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by nicosomb — APPROVED** [review]

**Review by boherm — APPROVED** [review]

File: src/PrestaShopBundle/Twig/Component/Toolbar.php [hunk 1]
```diff
@@ -28,19 +28,45 @@
...
+            $tabs[] = $tab;
+            $ancestorsTab = $this->menuBuilder->getAncestorsTab($tab->getId());
+            if (!empty($ancestorsTab)) {
+                $tabs[] = $ancestorsTab;
+            }
+
+            $this->breadcrumbs = $this->menuBuilder->convertTabsToBreadcrumbLinks($tab, $ancestorsTab);
+            $this->hookDispatcher->dispatchWithParameters('actionAdminBreadcrumbModifier', ['tabs' => $tabs, 'breadcrumb' => &$this->breadcrumbs]);
+        }
+    }
```
Comment (PululuK) [inline src/PrestaShopBundle/Twig/Component/Toolbar.php:71]:
```suggestion
    public array $breadcrumbs = [];
    public bool $useRegularH1Structure = true;

    public function __construct(
        private readonly HookDispatcherInterface $hookDispatcher,
        private readonly MenuBuilder $menuBuilder
    ) {
    }

    public function mount(): void
    {
        $tab = $this->menuBuilder->getCurrentTab();
        if ($tab) {
            $tabs = [];
            $tabs[] = $tab;
            $ancestorsTab = $this->menuBuilder->getAncestorsTab($tab->getId());
            if (!empty($ancestorsTab)) {
                $tabs[] = $ancestorsTab;
            }

            $this->breadcrumbs = $this->menuBuilder->convertTabsToBreadcrumbLinks($tab, $ancestorsTab);
            $this->hookDispatcher->dispatchWithParameters('actionAdminBreadcrumbModifier', ['tabs' => $tabs, 'breadcrumb' => &$this->breadcrumbs]);
        }
    }
```

**Review by PululuK — APPROVED** [review]
Look good for me ... small feedback, but not mandatory :)

File: src/PrestaShopBundle/Twig/Component/Toolbar.php (hunk 1 above)
Comment (M0rgan01) [reply src/PrestaShopBundle/Twig/Component/Toolbar.php:71]: Thanks for the feedback :) I made the same changes on another branch, based on this one and continuing the migration.

**Review by MhiriFaten — APPROVED** [review]
Hello @M0rgan01 ,

CI and auto tests are OK
https://github.com/MhiriFaten/testing_pr/actions/runs/5948074026

**Comment by nicosomb** [comment]
Thank you @M0rgan01 !

**Comment by jolelievre** [comment]
Tests with symfony layout https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/5955200145

**Pull request merged**